lib/modules/filteReddit.js

```diff
@@ -786,18 +801,224 @@ module.go = function() {
+		for (const p of parts) {
+			if (acceptedParts.length >= 8) break;
+			const part = decodeURIComponent(p);
+			if (!whitelist.test(part)) return null;
+			else acceptedParts.push(part);
 		}
+
+		if (!acceptedParts) return null;
+
+		const regex = new RegExp(`[\\[\\{\\<\\(]\\s*(${acceptedParts.join('|')})\\s*[\\]\\}\\>\\)]`, 'i');
+
+		return {
+			name: `subredditDefined-${label}`,
+			displayName: label,
+			test: thing => regex.test(thing.getPostLink().text),
+		};
+	}).filter(v => v);
+}
+
+function createFilterline() {
+	filterline = new Filterline();
+
+	const filters = [
+		{
+			name: 'nsfw',
+			test: module.options.customFilters.cases.isNSFW.evaluate,
+			visible: !module.options.NSFWfilter.value,
+		}, {
+			/* Not available for Safari and Safari, as they lack isURLVisited */
+			name: 'visited',
+			test: thing => isURLVisited(thing.getPostLink().href),
+			async: true,
+			visible: process.env.BUILD_TARGET !== 'safari' && process.env.BUILD_TARGET !== 'edge',
+		}, {
+			name: 'has expando',
+			command: /^ex(?:pando)?/,
+			description: 'ex[pando] [space separated types] filter by expando types',
+			test(thing) {
+				const expando = thing.getEntryExpando();
+				if (this.criterion && expando) {
+					const specified = (this.criterion || '').toUpperCase().split(' ');
+					return !!_.intersection(expando.getTypes(), specified).length;
+				} else {
+					return !!expando;
+				}
+			},
+			onChange() {
+				if (this.criterion) {
+					// showImages will not complete expandos if they are hidden, but if we have criterion
+					// we must know the expando type, which requires completion
+					forceExpandoCompletion = this.state;
+				}
+				// refresh the things separately when completed
+				expandoCompletionCallback = this.state === null ? null : refreshThingFilter;
+			},
+		}, {
+			name: 'post age',
+			command: 'age',
+			description: 'age [x(Y|M|d|h|m)] - less than x age',
+			test(thing) {
+				let age = parseFloat(this.criterion, 10) * 1000;
+				const ageQualifier = _.head(this.criterion.match(/Y|M|d|h|m/));
+
+				if (isNaN(age) || !ageQualifier) {
+					return null;
+				}
+
+				/* eslint-disable no-fallthrough, default-case */
+				switch (ageQualifier) {
+					case 'Y': age *= 12;
+					case 'M': age *= 30.44;
+					case 'd': age *= 24;
+					case 'h': age *= 60;
+					case 'm': age *= 60;
+				}
+				/* eslint-enable no-fallthrough, default-case */
+
+				return module.options.customFilters.cases.postAge.evaluate(thing, { op: '<', age });
+			},
+			visible: false,
+			visibleOnCriterion: true,
+		}, {
+			name: 'score',
+			command: 'score',
+			description: 'score [x] - higher than x score',
+			test(thing) {
+				const score = parseInt(this.criterion, 10);
+				if (isNaN(score)) return null;
+				return module.options.customFilters.cases.score.evaluate(thing, { op: '>', val: score });
+			},
+			visible: false,
+			visibleOnCriterion: true,
+		},
+	];
+
+	filters.forEach(::filterline.addFilter);
```

Comment:
We don't need to bind these; I'd prefer `filters.forEach(filter => filterline.addFilter(filter))` or a for-of loop